cds/modules/opencast/tasks.py

```diff
@@ -195,54 +214,49 @@ def check_transcoding_status():
             )
         )
     for started_flow_tasks in grouped_flow_tasks:
-
         # the opencast event id is the same for all transcoding tasks
         event_id = started_flow_tasks[0].payload["opencast_event_id"]
         opencast_event = opencast_events.get(event_id)
         if not opencast_event:
             continue
 
-        for started_flow_task in started_flow_tasks:
-            for subformat in opencast_event["subformats"]:
-                is_same_preset_quality = (
-                    started_flow_task.payload["opencast_publication_tag"]
-                    in subformat["tags"]
-                )
-                if not is_same_preset_quality:
-                    continue
-
-                on_transcoding_completed.s(
-                    flow_task_id=str(started_flow_task.id),
-                    opencast_subformat=subformat,
-                    opencast_event_id=event_id
-                ).apply_async(
-                    link_error=on_celery_task_failed.s(
-                        data=dict(flow_task_id=str(started_flow_task.id))
-                    ),
-                )
-                number_of_transcoding_tasks_started += 1
-                # transcoding subformat completed, `break` to go to the next
-                break
-            else:
-                # no `break`, the transcoded subformat might not yet
-                # ready, or if workflow failed, this subformat failed
-                if opencast_event["status"] == "FAILED":
-                    msg = (
-                        "Opencast event 'processing_state' field has "
-                        "value 'FAILED' for event id: {0}.".format(event_id)
-                    )
-                    _set_flow_tasks_to_failed(
-                        [(str(started_flow_task.id), msg)]
-                    )
-    if number_of_transcoding_tasks_started:
-        current_app.logger.debug(
-            "Started {0} on_transcoding_completed tasks".format(
-                number_of_transcoding_tasks_started
+        is_failed = opencast_event["processing_state"] == "FAILED"
+        if is_failed:
+            msg = (
+                "Opencast event 'processing_state' field has "
+                "value 'FAILED' for event id: {0}.".format(event_id)
             )
+            for started_flow_task in started_flow_tasks:
+                _set_flow_tasks_to_failed(
```

Comment:
Question: Before, we were only setting the tasks to failed if they were not present in the pauload and the status was FAILED. Now it looks like we only check for the status FAILED, not really checking that the subformat is in the payload. Does this mean that from now on, if the status is FAILED no subformats were processed in opencast, or could it be that it's FAILED and that some subformats were actually processed?

Reply:
Valid point, we cannot trust upstream on this, I will put it back.